src/portfolio/APIClient.ts

```diff
@@ -384,6 +447,30 @@ export async function list(token: string): Promise<PortfolioOverview[]> {
   ];
 }
 
+/**
+ * Gets the data needed for the Back-Testing section of the portfolio details page.
+ *
+ * @param token - Authentication token
+ * @param id - Id of the portfolio for which the Back-Testing information is needed
+ * @param from - UNIX timestamp for the date FROM which the user wants to backtest
+ * @param to - UNIX timestamp for the date UNTIL WHICH the user wants to backtest
+ */
+export async function backtesting(
+  token: string,
+  id: string,
+  from: Date,
+  to: Date
+): Promise<Backtesting> {
+  const response = (await request(
+    token,
+    'GET',
+    // TODO edit url when final contract is clear
+    `backtest/${id}/${from}/${to}`
+    // TODO add type BacktestingResponse if needed
+  )) as BacktestingResponse;
+  return convertBacktesting(response);
+}
```

Comment:
The API [documentation](https://api.milou.io/api-docs/#/analytics/get_analytics_backtest__portfolioId___fromDate___toDate_) has been updated. There are no responses with mock values, but the parameters and response format is already defined.

The dates are query parameters: `/analytics/backtest/{portfolioId}?{fromDate}&{toDate}`

The response body looks like this:
```json
{
  "error": "Portfolio with ID: 0 was not found!",
  "success": {}
}
```

Apparently, `error` and `success` are always present. On success, error should be an empty string.

Reply:
I updated backtesting(), BacktestingResponse and the convertBacktesting method. Could you check it out and see if it makes sense? Also could I have done the convert method in a cleaner way? I didn't know how to properly handle this "2-level-deep" object destructuring.

Reply:
See my new comments.